app/src/test/java/com/example/campusguide/utils/MessageDialogFragmentTest.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.campusguide.utils
+
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentTransaction
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.mockito.ArgumentMatchers.anyString
+import java.lang.IllegalStateException
+
+@RunWith(JUnit4::class)
+class MessageDialogFragmentTest {
+    @Test(expected = IllegalStateException::class)
+    fun testActivityNull() {
+        val dialog = MessageDialogFragment("someMessage")
+        val manager: FragmentManager = mock()
+        val transaction: FragmentTransaction = mock()
+        whenever(transaction.add(any(), anyString())).thenReturn(transaction)
+        whenever(transaction.commit()).thenReturn(-1)
+
+        whenever(manager.beginTransaction()).thenReturn(transaction)
+        dialog.show(manager, "someTag")
+        dialog.onCreateDialog(null)
```

Comment:
This doesn't seem to have any asserts?

Reply:
This test verifies that a message dialog fragment throws an IllegalStateException when created outside of an activity. Note the annotation @Test(expected = IllegalStateException::class) - the test will fail if the exception is not thrown.